src/server/claimQueue/claimQueueAPI.js

```diff
@@ -82,8 +82,48 @@ const ClaimQueue = {
     return { ok: 1, newAllowed, stillPending, approvedUsers: pendingUsers }
   },
 
-  async enqueue(user, storage, log) {
+  async updateAllowedEmails(emailsToApprove: Array<string>, storage, log) {
     const { claimQueueAllowed } = conf
+    const emailsHashes = map(emailsToApprove, sha3)
+    const approvedUsers = await storage.model
+      .find(
+        {
+          email: { $in: emailsHashes },
+          $or: [
+            { 'claimQueue.status': { $nin: ['approved', 'whitelisted'] } },
+            { 'claimQueue.status': { $exists: false } }
+          ]
+        },
+        { mauticId: 1, identifier: 1 }
+      )
+      .lean()
+
+    //update the global allowed so approved by email users dont take existing open spots
+    let queueProps = await ClaimQueueProps.findOne({})
+    if (!queueProps) {
+      queueProps = new ClaimQueueProps({ value: claimQueueAllowed })
+    }
+    queueProps.value += approvedUsers.length
+    await queueProps.save()
+
+    const userIds = map(approvedUsers, '_id')
+    const mauticIds = map(approvedUsers, 'mauticId')
+
+    Mautic.addContactsToSegment(mauticIds, conf.mauticClaimQueueApprovedSegmentId).catch(e => {
+      log.error('Failed Mautic adding user to claim queue approved segment', e.message, e)
+    })
+
+    await storage.model.updateMany({ _id: { $in: userIds } }, { $set: { 'claimQueue.status': 'approved' } })
+
+    log.debug('claim queue updated', { approvedUsers })
+    return { ok: 1, approvedUsers }
+  },
+
+  async enqueue(user, storage, log) {
+    const { claimQueueAllowed: claimQueueAllowedDefault } = conf
+
+    let queueProps = await ClaimQueueProps.findOne({})
+    const claimQueueAllowed = get(queueProps, 'value', claimQueueAllowedDefault)
```

Comment:
```suggestion
    // update the global allowed so approved by email users dont take existing open spots
    const queueProps = await this.getQueueProps()
    
    queueProps.value += approvedUsers.length
    await queueProps.save()

    const userIds = map(approvedUsers, '_id')
    const mauticIds = map(approvedUsers, 'mauticId')

    Mautic.addContactsToSegment(mauticIds, conf.mauticClaimQueueApprovedSegmentId).catch(e => {
      log.error('Failed Mautic adding user to claim queue approved segment', e.message, e)
    })

    await storage.model.updateMany({ _id: { $in: userIds } }, { $set: { 'claimQueue.status': 'approved' } })

    log.debug('claim queue updated', { approvedUsers })
    return { ok: 1, approvedUsers }
  },
  
  async getQueueProps() {
    const { claimQueueAllowed } = conf
    let queueProps = await ClaimQueueProps.findOne({})
    
    if (!queueProps) {
      queueProps = new ClaimQueueProps({ value: claimQueueAllowed })
      await queueProps.save()
    }

    return queueProps
  },

  async enqueue(user, storage, log) {
    const claimQueueAllowed = await this.getQueueProps().then(_ => _.value)
```

Reply:
also ClaimQueue could be converted to the class as have now a lot of methods some of them are internal:

```javascript
const ClaimQueue = new class {
  // ..
}()
```